backend/src/packages/chaiNNer_standard/utility/value/formats.py

```diff
@@ -0,0 +1,119 @@
+                ResList.NTSC_MPEG2: "SD NTSC MPEG-2 DVD 704x480",
+                ResList.NTSC: "SD NTSC 720x486",
+                ResList.NTSC_DV: "SD NTSC DV 720x480",
+                ResList.PAL: "SD PAL 720x576",
+                ResList.PAL_MPEG2: "SD PAL MPEG-2 DVD 702x576",
+                ResList.HD_720: "HD 720P 1280x720",
+                ResList.UHD_4K: "UHD 4K 3840x2160",
+                ResList.UHD2_8K: "UHD2 8K 7680x4320",
+                ResList.DCI_2KFLAT: "2K DCI FLAT 1998x1080",
+                ResList.DCI_2KSCOPE: "2K DCI SCOPE 2048x858",
+                ResList.DCI_2KFULL: "2K DCI FULL 2048x1080",
+                ResList.DCI_4KFULL: "4K DCI FULL 4096x2160",
+                ResList.DCI_4KFLAT: "4K DCI FLAT 3996x2160",
+                ResList.DCI_4KSCOPE: "4K DCI SCOPE 4096x1716",
+                ResList.SQ256: "Square 256x256",
+                ResList.SQ512: "Square 512x512",
+                ResList.SQ1024: "Square 1024x1024",
+                ResList.SQ2048: "Square 2048x2048",
+                ResList.SQ4096: "Square 4096x4096",
+                ResList.SQ8192: "Square 8192x8192",
+            },
+        ),
+    ],
+    outputs=[
+        NumberOutput("Width", output_type="1.."),
+        NumberOutput("Height", output_type="1.."),
+    ],
+)
+def formats_node(resolution_presets: ResList) -> tuple[int, int]:
+    if resolution_presets == ResList.HD_1080:
+        return 1920, 1080
+    elif resolution_presets == ResList.NTSC_MPEG2:
+        return 704, 480
+    elif resolution_presets == ResList.NTSC:
+        return 720, 486
+    elif resolution_presets == ResList.NTSC_DV:
+        return 720, 480
+    elif resolution_presets == ResList.PAL:
+        return 720, 576
+    elif resolution_presets == ResList.PAL_MPEG2:
+        return 702, 576
+    elif resolution_presets == ResList.HD_720:
+        return 1280, 720
+    elif resolution_presets == ResList.UHD_4K:
+        return 3840, 2160
+    elif resolution_presets == ResList.UHD2_8K:
+        return 7680, 4320
+    elif resolution_presets == ResList.DCI_2KFULL:
+        return 2048, 1080
+    elif resolution_presets == ResList.DCI_2KFLAT:
+        return 1998, 1080
+    elif resolution_presets == ResList.DCI_2KSCOPE:
+        return 2048, 858
+    elif resolution_presets == ResList.DCI_4KFULL:
+        return 4096, 2160
+    elif resolution_presets == ResList.DCI_4KFLAT:
+        return 3996, 2160
+    elif resolution_presets == ResList.DCI_4KSCOPE:
+        return 4096, 1716
+    elif resolution_presets == ResList.SQ256:
+        return 256, 256
+    elif resolution_presets == ResList.SQ512:
+        return 512, 512
+    elif resolution_presets == ResList.SQ1024:
+        return 1024, 1024
+    elif resolution_presets == ResList.SQ2048:
+        return 2048, 2048
+    elif resolution_presets == ResList.SQ4096:
+        return 4096, 4096
+    elif resolution_presets == ResList.SQ8192:
+        return 8192, 8192
+    else:
+        return 1920, 1080
```

Comment:
I think it would be better to define the resolutions in a dict and then to look them up. This will eliminate this long `elif` chain, and allows us to easily check that we didn't forget a format (`assert len(RESOLUTION) == len(ResList)`).

Reply:
Have an example node where this is already setup for me to look at? The elif will only grow as we add more resolutions in the future so I’d be interested in a more elegant approach. 

I’d also love to be able to do cascading menus in chaiNNer in an easier way for an even bigger “camera resolutions node” in the future which could have hundreds of presets broken down per camera.

Reply:
I think from a chaiNNer node design/Ux paradigm having an easy means to do cascading menus would be a great addition. For now though I'll do my best attempt to learn some more Python fu and get this dict thing to work!

Reply:
Done, moved 99% of the presets to a dict.